include/xdg_shell_v6.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef WLCS_XDG_SHELL_V6_
+#define WLCS_XDG_SHELL_V6_
+
+#include "in_process_server.h"
+#include "generated/xdg-shell-unstable-v6-client.h"
+
+namespace wlcs
+{
+
+class XdgSurfaceV6
+{
+public:
+    XdgSurfaceV6(wlcs::Client& client, wlcs::Surface& surface);
+    ~XdgSurfaceV6();
+
+    void add_configure_notification(std::function<void(uint32_t)> notification)
+    {
+        configure_notifiers.push_back(notification);
+    }
+
+    operator zxdg_surface_v6*() const {return shell_surface;}
+
+    wlcs::Client* const client;
+    wlcs::Surface* const surface;
+    zxdg_surface_v6* shell_surface;
```

Comment:
Nit: This should be private as non-members should not update it. Consider a conversion operator for non-member access?

Reply:
Done